dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/CompanyRolesControllerTest.kt

```diff
@@ -282,4 +288,123 @@ class CompanyRolesControllerTest {
+        assignCompanyRole(CompanyRole.CompanyOwner, companyId, dataReaderUserId)
+
+        enumValues<CompanyRole>().forEach {
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+            assignCompanyRole(it, companyId, dataUploaderUserId)
+
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+            assertDoesNotThrow { hasUserCompanyRole(it, companyId, dataUploaderUserId) }
+
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+            removeCompanyRole(it, companyId, dataUploaderUserId)
+
+            val exceptionWhenCheckingIfUserIsCompanyOwner = assertThrows<ClientException> {
+                hasUserCompanyRole(it, companyId, dataUploaderUserId)
+            }
+            assertErrorCodeInCommunityManagerClientException(exceptionWhenCheckingIfUserIsCompanyOwner, 404)
+        }
+    }
+
+    @Test
+    fun `assure that a company member admin without admin rights can only add and remove members and member admins`() {
+        val companyId = uploadCompanyAndReturnCompanyId()
+        val listOfRolesThatCanBeModified = listOf(CompanyRole.MemberAdmin, CompanyRole.Member)
+        val listOfRolesThatCannotBeModified =
+            listOf(CompanyRole.CompanyOwner, CompanyRole.DataUploader)
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Admin)
+        assignCompanyRole(CompanyRole.MemberAdmin, companyId, dataReaderUserId)
+        listOfRolesThatCanBeModified.forEach {
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+            assignCompanyRole(it, companyId, dataUploaderUserId)
+
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Uploader)
+            assertDoesNotThrow { hasUserCompanyRole(it, companyId, dataUploaderUserId) }
+
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+            removeCompanyRole(it, companyId, dataUploaderUserId)
+
+            val exceptionWhenCheckingIfUserIsCompanyOwner = assertThrows<ClientException> {
+                hasUserCompanyRole(it, companyId, dataUploaderUserId)
+            }
+            assertErrorCodeInCommunityManagerClientException(exceptionWhenCheckingIfUserIsCompanyOwner, 404)
+        }
+
+        listOfRolesThatCannotBeModified.forEach {
+            jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.Reader)
+            val exceptionWhenTryingToAddCompanyMembers = assertThrows<ClientException> {
+                assignCompanyRole(it, companyId, dataUploaderUserId)
+            }
+            assertErrorCodeInCommunityManagerClientException(exceptionWhenTryingToAddCompanyMembers, 403)
+        }
+    }
+
+    @Test
+    fun `assure that a user that have member or uploader role or no role at all cannot add or remove company roles`() {
```

Comment:
Lass uns den Text auch leicht umändern, dass glasklar ist, dass du von company-roles sprichst.

Z.B. `assure that modifying role assignments is not possible for user with no company role or only member or uploader company role` 

oder sowas

Reply:
Done

Reply:
Slightly rearranged the test name so that it is compliant with detekt rules